pkg/cli/admin/inspect/pod.go

```diff
@@ -17,7 +17,7 @@ import (
 )
 
 func (o *InspectOptions) gatherPodData(destDir, namespace string, pod *corev1.Pod) error {
-	if pod.Status.Phase != corev1.PodRunning {
+	if pod.Status.Phase != corev1.PodRunning && !o.inspectNonRunningPod {
```

Comment:
That's not sufficient, we scrape much more data, which rely on the pod running. You'd need a clear definition what you can scrape from running and what from other pods. The current condition changes are not sufficient.